### PR TITLE
Add docs to pipeline units and their usage

### PR DESCRIPTION
**Pull request by fridex**

### CODE REVIEW

File: docs/source/unit.rst [hunk 1]
```diff
@@ -150,3 +150,52 @@ It's good to note how pipeline units should be listed in ``__all__``:
   that performs a knowledge graph query (the more queries or more expensive
   queries, the more pipeline unit is expensive). Note the overhead needed to
   query the knowledge base.
+
+Which pipeline unit type should be chosen?
+==========================================
+
+Sometimes it might be tricky to select the right pipeline unit. There were
+designed multiple unit types to provide a framework for resolver to easily
```
Comment (KPostOffice) [inline docs/source/unit.rst:158]: Multiple unit types were designed to provide*

File: docs/source/unit.rst [hunk 2]
```diff
@@ -150,3 +150,52 @@ It's good to note how pipeline units should be listed in ``__all__``:
...
+write units. These units have different overhead and are designed for specific
+use cases. It's crucial to select the right pipeline unit for the right use
+case to keep the pipeline well performing.
```
Comment (KPostOffice) [inline docs/source/unit.rst:161]: Performing well*

File: docs/source/unit.rst [hunk 3]
```diff
@@ -150,3 +150,52 @@ It's good to note how pipeline units should be listed in ``__all__``:
...
+
+The most expensive pipeline units are :ref:`steps <steps>`. They are run each
+time a package is about to be added to resolver's internal state. As it is the
+most expensive one, it also provides the most information for a pipeline unit
+developer - which package in which specific version is about to be added to a
+partially resolved state and how the resolver state looks like. These units, as
```
Comment (KPostOffice) [inline docs/source/unit.rst:167]: what* the resolver state looks like

Comment (KPostOffice) [inline docs/source/unit.rst:167]: These units are the only ones that can affect the final unit score.

File: docs/source/unit.rst [hunk 4]
```diff
@@ -150,3 +150,52 @@ It's good to note how pipeline units should be listed in ``__all__``:
...
+the only ones, can affect the final unit score.
+
+The second most expensive pipeline units are :ref:`sieves <sieves>`. They do
+not provide access to resolver's internal state, but are called each time there
+are packages in specific versions considered for further resolution. As the
+name suggests, these units filter out packages that should not occur in the
+final software stack. These units, in opposite to :ref:`steps <steps>`, do not
```
Comment (KPostOffice) [inline docs/source/unit.rst:174]: unlike* steps

File: docs/source/unit.rst [hunk 5]
```diff
@@ -150,3 +150,52 @@ It's good to note how pipeline units should be listed in ``__all__``:
...
+final changes to final states that are not relevant to the state score,
+packages resolved in the final state or resolver input vector.
+
+Refer to sections specific to pipeline unit types for examples and more
+information.
+
+Unroll pipeline units
+=====================
+
+To keep the resolver well-performing, try to always unroll all the operations
```
Comment (KPostOffice) [inline docs/source/unit.rst:195]: performing well*

**Comment by fridex** [comment]
@KPostOffice thanks, review comments addressed. PTAL :+1:

**Pull request merged**